server/request/request.go

```diff
@@ -56,11 +51,17 @@ type Metadata struct {
 	methodInfo  grpc.MethodInfo
 	// this is authenticated namespace
 	namespace string
-	IsHuman   bool
+	// human readable namespace name
+	namespaceName string
+	IsHuman       bool
 	// this is parsed and set early in request processing chain - before it is authenticated for observability.
 	unauthenticatedNamespaceName string
 }
 
+func Init(tm *metadata.TenantManager) {
+	tenantMgr = tm
```

Comment:
Instead of directly depending on the metadata package, can we have an interface in this package to access Tenant object?
```
type TenantGetter interface {
	Get(ctx context.Context, id string) (Tenant, error)
}
```

Reply:
Sounds great, I changed it like this.